src/openrct2/ride/Vehicle.cpp

```diff
@@ -4339,12 +4339,13 @@ void Vehicle::TryReconnectBoatToTrack(const CoordsXY& currentBoatLocation, const
         TrackLocation.x = trackCoords.x;
         TrackLocation.y = trackCoords.y;
 
-        auto trackElement = map_get_track_element_at(TrackLocation);
-
         auto curRide = GetRide();
         if (curRide != nullptr)
         {
-            SetTrackType(trackElement->GetTrackType());
+            auto trackElement = map_get_track_element_at(TrackLocation);
+            if (trackElement != nullptr)
+                SetTrackType(trackElement->GetTrackType());
+
```

Comment:
not sure this is safe to do. If it fails to reconnect then setting the boat location to null and not setting the track type will cause other issues. I think perhaps this if != nullptr should encompass the same amount of scope as the curRide check.

Reply:
Is `BoatLocation` null if the boat is on a guide track, and set if it's floating freely?

Reply:
Also, if you take a look at git blame, the BoatLocation.setNull() call was put inside the if because it was split off from trackdirection, and that was only inside the if to prevent a NPE when `ride` was null.

Still, I'm not sure what the "safest" thing to do is here: leave it alone, or set it to a fail-safe value.

Reply:
Hmm not sure what would be best

Reply:
So do we close or merge this @duncanspumpkin ?

Reply:
I'm still not sure. We could just make the boat explode. Its the easiest get out for this...